src/Prometheus.Client.HttpClient/MessageHandlers/HttpClientInProgressHandler.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Prometheus.Client.HttpClient.MessageHandlers
+{
+    public sealed class HttpClientInProgressHandler : HttpClientMessageHandlerBase<IMetricFamily<IGauge, ValueTuple<string, string, string, string>>, IGauge>
+    {
+        public HttpClientInProgressHandler(IMetricFactory metricFactory, string clientName)
+            : base(metricFactory, clientName)
+        {
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            HttpResponseMessage response = null;
+            var metric = CreateMetric(request, null);
+
+            metric.Inc();
+
+            try
+            {
+                response = await base.SendAsync(request, cancellationToken);
+            }
+            finally 
+            {
+                metric.Dec();
+            }
+
+            return response;
+        }
+
+        protected override IMetricFamily<IGauge, ValueTuple<string, string, string, string>> CreateMetricInstance() => MetricFactory.CreateGauge(Constants.InProgresMetricName, Constants.InProgresMetricHelp, (Constants.Host, Constants.Client, Constants.Method, Constants.StatusCode));
```

Comment:
The next feature will be Custom Metric name, I recommend move `MetricFactory.CreateGauge` to constructor.

Reply:
@phnx47 
The plan is to have options in handler classes to get the `customMetricName`.